File: php/features.php
<?php
declare(strict_types=1);

// PhoneNumberValidator SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PhoneNumberValidatorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PhoneNumberValidatorBaseFeature();
            case "test":
                return new PhoneNumberValidatorTestFeature();
            default:
                return new PhoneNumberValidatorBaseFeature();
        }
    }
}

File: py/features.py
# PhoneNumberValidator SDK feature factory

from feature.base_feature import PhoneNumberValidatorBaseFeature
from feature.test_feature import PhoneNumberValidatorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PhoneNumberValidatorBaseFeature(),
        "test": lambda: PhoneNumberValidatorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
